src/git.rs

```diff
@@ -14,9 +14,9 @@ type Result = result::Result<(), String>;
 
 pub trait GitWrapper {
     fn git_version(&self) -> Result;
-    fn git_clone_mirror(&self, origin: &str, repo_dir: &PathBuf) -> Result;
-    fn git_update_mirror(&self, origin: &str, repo_dir: &PathBuf) -> Result;
-    fn git_push_mirror(&self, dest: &str, repo_dir: &PathBuf) -> Result;
+    fn git_clone_mirror(&self, origin: &str, repo_dir: &PathBuf, branch: &str) -> Result;
+    fn git_update_mirror(&self, origin: &str, repo_dir: &PathBuf, branch: &str) -> Result;
+    fn git_push_mirror(&self, dest: &str, repo_dir: &PathBuf, branch: &str) -> Result;
```

Comment:
Like mentioned bellow, maybe thos should be an `Option<&str>` instead of doing the decision on empty string.